### ISSUE TITLE
getattr- getattr : signature does not have a field dayOfWeek

### ISSUE THREAD

**Issue by hankcs** [telepathylabsai/OpenDF#19]
Dear authors,

I found OpenDF is trying to repeatedly call  `getattr` while the signature validation failed:

## Dialog 54

**GUID**: `096db3b5-b0cf-4268-8841-04936f86ac2b` **OpenDF**:

 - U1: Did I host a graduation party last year?
 - A1: Yes, I found one event matching "graduation party". It was "Graduation Party" on May 19, 2019 from 5:00 to 5:30 PM.
 - U2: What did I do **after the** graduation party last year?
 - A2: I found one event on May 19, 2019 after 5:30 PM. It was "Graduation After-Party" from 7:00 to 7:30 PM.
 - U3: What day was the graduation party event last year?
 - A3: On May 19, 2019, it was Sunday.

Program:

```
 :dayOfWeek(
    :date(
       :start(
          FindEvents(
             AND(
                at_time(
                   FullYearofYear(
                      LastYear(
                         ) ) ) ,
                has_subject(
                   graduation party ) ) ) ) ) ) 
```

Graph:
![Screen Shot 2022-06-21 at 11 42 15 AM](https://user-images.githubusercontent.com/5326890/174841373-8b579f13-cbe6-42c9-8e4d-f482daebb525.png)

**Comment by meron-tl** [comment]
The cause was that dayOfWeek is not an input of Date. fixed in commit e7cc2dbd

**Comment by hankcs** [comment]
Fix confirmed, thank you.
